### ISSUE TITLE
Latest Chrome broke `fly login` UI response

### ISSUE THREAD

**Issue by Esysc** [concourse/concourse#8529]
## Summary

It is not strictly related to Concourse and it is a minor issue related to user experience.
Since Google Chrome `v104` the 2 ways communication between `fly` and the web browser is half broken. Actually the POST request to the local ephemeral port is refused and the UI shows the information pop-up that the login is successful, but the token could not be sent to `fly`.
With older version of Chrome, for example `101.0.4951.41`, it behaves correctly.
With latest Firefox `v103.0` it behaves correctly too.

## Steps to reproduce

- Logout from all targets: `fly logout --all`
- Login to a target: `fly -t main login`
- Open the provided link with Chrome

## Expected results
Login successful and pop-up message below:
![image](https://user-images.githubusercontent.com/9631930/185373203-22d46633-0f39-4b16-ab47-bcc899bb4d42.png)

## Actual results

login successful and pop-up message below:
![image](https://user-images.githubusercontent.com/9631930/185373336-3c57ec3c-54d5-44cc-ac4b-6ab5b5ed75d7.png)

## Additional context

<details>
<summary>Failed request</summary>

```
Request URL: http://127.0.0.1:39245/?token=bearer%20B%2BBXm1hQAX%2B....
Referrer Policy: strict-origin-when-cross-origin
Accept: */*
Accept-Encoding: gzip, deflate, br
Accept-Language: en,en-GB;q=0.9,en-US;q=0.8,fr;q=0.7
Connection: keep-alive
Host: 127.0.0.1:39245
Origin: https://<concourse>
sec-ch-ua: "Chromium";v="104", " Not A;Brand";v="99", "Google Chrome";v="104"
sec-ch-ua-mobile: ?1
sec-ch-ua-platform: "Android"
Sec-Fetch-Dest: empty
Sec-Fetch-Mode: cors
Sec-Fetch-Site: cross-site
User-Agent: Mozilla/5.0 (Linux; Android 6.0; Nexus 5 Build/MRA58N) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.0.0 Mobile Safari/537.36
```

![image](https://user-images.githubusercontent.com/9631930/185375326-dd730c9e-3b55-4df9-a597-27872dfc9aef.png)

</details>

## Triaging info

* Concourse version: 7.8.1
* Browser (if applicable): Google Chrome `104.0.5112.79`
* Did this used to work? Yes

**Comment by taylorsilva** [comment]
Some additional details from testing on Chrome 104.

- Fly DOES get the token. Login is successful and I don't need to paste the token over to fly.
- It appears the same request is sent twice. The second request fails because fly is no longer listening for the token after it gets it from the first request 
  ![image](https://user-images.githubusercontent.com/4583837/186204137-f7f6f42b-0d91-47ef-b3f8-dd6d9c53fd7f.png)

@Esysc Can you confirm that the above is true for you too?

I personally don't know a quick fix for this since it seems like a bug somewhere in the Elm code which I'm not very familiar with.

**Comment by Esysc** [comment]
@taylorsilva you are right,  but it looks to me a change on how chrome handles this. When I'll have some spare time I'll investigate more

**Comment by taylorsilva** [comment]
Ended up seeing this in the chrome dev tools: <img width="1603" alt="Image" src="https://github.com/user-attachments/assets/aa310cd8-38cf-49a1-8e69-9c9b6c2c0be8" />

Had fly add the header but issue persists. Still digging into this.

**Comment by taylorsilva** [comment]
`tailscale cert` to the rescue! https://tailscale.com/kb/1153/enabling-https

**Comment by taylorsilva** [comment]
Found what broke this whole thing in Chrome: https://developer.chrome.com/blog/private-network-access-preflight/

So Chrome is sending TWO requests, a "preflight" and then the actual request. So there's nothing our frontend code can do to handle this because the browser just _decides_ to send two requests when we _think_ we're only sending one.

This is the key part:

> Chrome will start sending a CORS preflight request ahead of any [private network request](https://developer.chrome.com/blog/private-network-access-preflight/#what_is_private_network_access_pna) for a subresource, which asks for explicit permission from the target server. This preflight request will carry a new header, `Access-Control-Request-Private-Network: true`, and the response to it must carry a corresponding header, `Access-Control-Allow-Private-Network: true`.

So this means we need to update `fly` to catch this preflight request if it detects it and wait around for the "real" second request, that way the Concourse frontend can actually know the request was successfully sent.

This is such a horrible way to implement this on Chrome's end.
